openedx/openedx-platform
# SE-2320 Adds django-cookies-sameseite middleware

Allows secure cookies to use SameSite=None, and sets secure cookies with secure requests.

The [SameSite cookie setting is supported in Django 2.1](https://github.com/django/django/blob/f436c82637dafa3a9abbd65a3be77bf7ad431213/docs/releases/2.1.txt#L109-L110), but [SameSite=None wasn't allowed until recently](https://github.com/django/django/pull/11894), so we need to use the `DCS_`-prefixed settings to allow this mode.

To support SameSite=None, we will need this middleware until Django>3.0.5.

**Discussions**

* Description of [Chrome's upcoming changes with SameSite and secure cookies](https://blog.heroku.com/chrome-changes-samesite-cookie)
   Note that Chrome has [delayed the release of this policy change](https://blog.chromium.org/2020/04/temporarily-rolling-back-samesite.html), but it is still coming.
* Community discussion on [Open edX forum](https://discuss.openedx.org/t/lti-xblock-and-samesite/759/6)

**Sandbox**

* LMS: https://pr23671.sandbox.opencraft.hosting/
* Studio: https://studio.pr23671.sandbox.opencraft.hosting/

**Testing instructions**

1. Visit the sandbox LMS, and inspect the browser cookies.
    Note that all the cookies are secure, and marked with "SameSite: None".
1. Register a new account or log in using one of the demo accounts.
    Note that all the cookies are secure, and marked with "SameSite: None".

**Reviewer**


FYI  @danielmcquillen @ztraboo  @sambapete 

**Settings**

```yaml
EDXAPP_CSRF_COOKIE_SECURE: true
EDXAPP_SESSION_COOKIE_SECURE: true
```
